### PR TITLE
#205 - Manage mousenter/mouseleave events

### PR DESCRIPTION
**Pull request by stfufane**
Hi there,

I know it was not the most urgent issue, but I wanted to give it a try to learn a bit more about the react side. 

I modified the GainPlugin example to have the button change its border color on hover, it works but I'll let you check if it's done correctly, I may have misunderstood the right way to do it.

Cheers!

### CODE REVIEW

**Comment by JoshMarler** [comment]
@stfufane, nailed it. This looks great. 

I'll pull down when I'm back at my desk to test but no issues from me. Anything to add @nick-thompson or shall I go ahead and merge ? Will do the package bumping necessaries.

File: packages/react-juce/src/components/Button.tsx [hunk 1]
```diff
@@ -84,14 +86,26 @@ export class Button extends Component<ButtonProps, ButtonState> {
     }
   }
 
+  handleEnter = (e: SyntheticMouseEvent): void => {
```
Comment (nick-thompson) [inline packages/react-juce/src/components/Button.tsx:89]: For cases like this one where we only wrap the supplied `props.onMouseEnter` in a function which checks that the thing is callable, we can just omit this code.

MouseDown, MouseUp, and onClick are special cases where we have to do some juggling for the end user before we call their provided callbacks. MouseEnter and MouseLeave are straightforward enough that we can just let them through to the underlying `View` if they're present.

So that means that where we write:
```
const { onMouseDown, onMouseUp, onClick, ...other } = this.props;
```
onMouseLeave and OnMouseEnter will get wrapped up into `...other` if they're present, and then get unwrapped via `{...other}` onto the View on line 111. If these props are not available, they won't get wrapped/unwrapped and passed on, which is just as we want.

So basically you can omit all the changes to this file except for 2 line change to the typescript interface above

**Review by nick-thompson — APPROVED** [review]
Agreed this is essentially perfect. Great work @stfufane thank you.

I left one comment in line where we can remove some code. After that, I think we're quite ready to merge!

**Comment by nick-thompson** [comment]
And yea @JoshMarler, after the quick edit to remove that bit of code above, if you get around to the merge before I do, yes please + thank you!

File: packages/react-juce/src/components/Button.tsx (hunk 1 above)
Comment (JoshMarler) [reply packages/react-juce/src/components/Button.tsx:89]: Ahh yeah good shout. Skimmed too quickly.

Comment (stfufane) [reply packages/react-juce/src/components/Button.tsx:89]: Oh okay I see, thanks for the review :) I noticed the more complex behaviour with the other events but didn't realize it wasn't needed to explicitly declare these ones, but the explanation is clear! 

I'll clean this and commit the changes!

Comment (stfufane) [reply packages/react-juce/src/components/Button.tsx]: It's done! I'll let you check by principle but I tested and it still works without the useless code :)

**Comment by nick-thompson** [comment]
Excellent, thanks @stfufane! I'll go ahead and rebase + merge this, then bump the npm package.

**Comment by nick-thompson** [comment]
Merged in a9f1f53cafc1ba4c982ef53d78416acbd99b8f30 and e12e99ec485c4593afdfac86f3a6f1494886a0b8 (commit hashes changed due to the rebase). I'll go ahead and close this accordingly
